packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1192,4 +1194,76 @@ class TextInput {
+  /// true.
+  ///
+  /// Currently, there can only be at most one autofill context at any given
+  /// time. When any input field in an [AutofillGroup] requests for autofill
+  /// (which is done automatically when an autofillable [EditableText] gains
+  /// focus), autofillable input fields in the group that may need saving will be
+  /// added to the existing autofill context, if it's not already in the context.
+  /// When there isn't an existing autofill context, one will be created to hold
+  /// the newly added input fields.
+  ///
+  /// Once added to an autofill context, an input field will stay in the context
+  /// until the context is destroyed. To prevent leaks, call [finishAutofillContext]
+  /// to signal the text input plugin that the user has finalized their input in
+  /// the current autofill context. The platform text input plugin either
+  /// encourages or discourages the platform from saving the user input based on
+  /// the value of the [shouldSave] parameter. The platform usually shows a
+  /// "Save for autofill?" prompt to ask for confirmation from the user for
+  /// saving the input.
+  /// {@endtemplate}
+  ///
+  /// Typically, this method should be called when the user has finalized their
+  /// input and ready to submit. For example, in a [Form], it's typically done
+  /// immediately before or after its content is submitted. You should make sure
+  /// that any connected [TextInputClient] is disconnected and no input field is
+  /// focused before calling [finishAutofillContext]. This is to ensure the user
+  /// can't interact with the input fields while it's being processed.
+  ///
+  /// In many cases you don't have to call [finishAutofillContext] yourself. The
+  /// [AutofillContextLifecycleAction] widget provides configurable event hooks
+  /// on [Navigator]s and [Form]s that you can call [finishAutofillContext] from.
+  /// By default, it calls [finishAutofillContext] with [shouldSave] equals to
+  /// `true` when a [Form] is saved, or whenever the an opaque [ModalRoute] in
+  /// any [Navigator]s becomes the current route, or is popped/removed from its
+  /// [Navigator]. See [AutofillContextLifecycleAction] for more details.
+  ///
+  /// Calling [finishAutofillContext] may cause the platform to show the "save
+  /// for autofill" UI, which should typically only be shown no more than once
+  /// for every screen. Consider removing premature [finishAutofillContext] calls
+  /// to prevent showing the "Save for autofill?" UI too frequently. However,
+  /// calling [finishAutofillContext] when there's no existing autofill context
+  /// does not do anything.
+  ///
+  /// ## Limitations
+  ///
+  /// Currently there can be at most one autofill context, and
+  /// [finishAutofillContext] either saves/discards everything or nothing at all.
+  /// With more complex navigation schemes, the ability to switch autofill
+  /// contexts is desirable.
```

Comment:
This feels a bit inconclusive.  Is there anything we can recommend for users to do who want multiple contexts?